pkg/controllers/provisioning/controller.go

```diff
@@ -118,6 +123,23 @@ func (c *Controller) Apply(ctx context.Context, provisioner *v1alpha5.Provisione
 	return nil
 }
 
+// Returns true if the new candidate provisioner is different than the provisioner in memory.
+func (c *Controller) hasChanged(ctx context.Context, provisionerNew *v1alpha5.Provisioner) bool {
+	oldProvisioner, _ := c.provisioners.Load(provisionerNew.Name)
+	if oldProvisioner == nil {
```

Comment:
I don't think it's safe to rely on the oldProvisioner being nil here. In go, the foo-or-error pattern has the semantic guarantee that the object will be set if the error is nil, but the reverse is not always true. I'd probably do something like 
```suggestion
	oldProvisioner, ok := c.provisioners.Load(provisionerNew.Name)
	if !ok {
```